public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -260,9 +261,18 @@ export class LokiDatasource
       .map(getNormalizedLokiQuery) // "fix" the `.queryType` prop
       .map((q) => ({ ...q, maxLines: q.maxLines ?? this.maxLines }));
 
+    const transformed = queries.map((query) => {
+      const sorted = sortLabelSelectors(query);
+
+      const lines = sorted.expr.split('\n');
+      const trimmedLines = lines.map((line) => line.trim());
+      const trimmedExpr = trimmedLines.join('\n');
+      return { ...sorted, expr: trimmedExpr };
+    });
+
```

Comment:
I would move all this to `queryUtils` and just call a function from there, so we don't keep piling up code in the datasource.